android/src/main/java/com/guichaguri/trackplayer/model/Track.kt

```diff
@@ -35,97 +36,33 @@ class Track(context: Context, bundle: Bundle, ratingType: Int) : TrackMetadata()
-//        if (userAgent == null || userAgent!!.isEmpty()) userAgent = Util.getUserAgent(
-//            ctx!!, "react-native-track-player"
-//        )
-//        val ds: DataSource.Factory?
-//        ds = if (resourceId != 0) {
-//            try {
-//                val raw = RawResourceDataSource(ctx!!)
-//                raw.open(DataSpec(uri!!))
-//                DataSource.Factory { raw }
-//            } catch (ex: IOException) {
-//                // Should never happen
-//                throw RuntimeException(ex)
-//            }
-//        } else if (Utils.isLocal(uri)) {
-//
-//            // Creates a local source factory
-//            DefaultDataSourceFactory(ctx!!, userAgent)
-//        } else {
-//
-//            // Creates a default http source factory, enabling cross protocol redirects
-//            val factory = DefaultHttpDataSourceFactory(
-//                userAgent, null,
-//                DefaultHttpDataSource.DEFAULT_CONNECT_TIMEOUT_MILLIS,
-//                DefaultHttpDataSource.DEFAULT_READ_TIMEOUT_MILLIS,
-//                true
-//            )
-//            if (headers != null) {
-//                factory.defaultRequestProperties.set(headers!!.toMap())
-//            }
-//            playback.enableCaching(factory)
-//        }
-//        return when (type) {
-//            TrackType.DASH -> createDashSource(ds)
-//            TrackType.HLS -> createHlsSource(ds)
-//            TrackType.SMOOTH_STREAMING -> createSsSource(ds)
-//            else -> ProgressiveMediaSource.Factory(
-//                ds, DefaultExtractorsFactory()
-//                    .setConstantBitrateSeekingEnabled(true)
-//            )
-//                .createMediaSource(uri!!)
-//        }
-//    }
-
-//    private fun createDashSource(factory: DataSource.Factory?): MediaSource {
-//        return DashMediaSource.Factory(DefaultDashChunkSource.Factory(factory!!), factory)
-//            .createMediaSource(uri!!)
-//    }
-//
-//    private fun createHlsSource(factory: DataSource.Factory?): MediaSource {
-//        return HlsMediaSource.Factory(factory!!)
-//            .createMediaSource(uri!!)
-//    }
-//
-//    private fun createSsSource(factory: DataSource.Factory?): MediaSource {
-//        return SsMediaSource.Factory(DefaultSsChunkSource.Factory(factory!!), factory)
-//            .createMediaSource(uri!!)
-//    }
-
     fun toAudioItem(): TrackAudioItem {
         return TrackAudioItem(
             this,
-            MediaType.DEFAULT,
+            type,
             uri.toString(),
             artist,
             title,
             album,
-            artwork.toString()
+            artwork.toString(),
+            AudioItemOptions(
+                headers,
+                userAgent,
+                resourceId
+            )
         )
     }
 
     init {
         resourceId = Utils.getRawResourceId(context, bundle, "url")
         uri = if (resourceId == 0) {
+            resourceId = null
```

Comment:
It looks like KotlinAudio expects this to be null, if its not to be used, so in a case that we're not using a local asset I set this back to null, so its not used when we call `toAudioItem`